### PR TITLE
Backport PR-60 to lunar-devel

### PR DESCRIPTION
**Pull request by jonbinney**
Add getFilters() function to provide access to FilterChain's filters
to child classes.

### CODE REVIEW

**Comment by peci1** [comment]
I tested this PR on my Melodic machine and it works as expected. Thanks for the backport!

**Comment by jonbinney** [comment]
Looks like this can't be merged without approval from another person with write access. @tfoote could you approve this if it looks ok? It is a straight backport of PR 60 to the lunar-devel branch, which is used for melodic. I think it shouldn't affect ABI or API because:
- adding a function shouldn't affect the ABI
- we now use std::shared_ptr instead of boost internally, but those only get allocated within a member vector; they won't affect the size of the class on the stack (even if std shared pointers and boost shared pointers were different sizes, which i doubt)

**Review by tfoote — APPROVED** [review]

**Pull request merged**
